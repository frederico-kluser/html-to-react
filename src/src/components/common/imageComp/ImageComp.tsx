/* eslint-disable @next/next/no-img-element */
import React from 'react';
import LazyLoad from 'react-lazyload';

type ImageCompProps = {
  src: string;
  alt?: string;
  height?: number;
  width?: number;
  isNext?: boolean;
  dataTestid?: string;
};

const ImageComp = (props: ImageCompProps) => {
  const { src, alt, height, width, isNext, dataTestid } = props;

  const chooseImageComp = () => {
    if (isNext) {
      // TODO switch HTML img to NextJS Image Component
      return (
        <img
          src={src}
          alt={alt}
          height={height}
          width={width}
          data-testid={dataTestid}
        />
      );
    }

    return (
      <LazyLoad height={height}>
        <img
          src={src}
          alt={alt}
          height={height}
          width={width}
          data-testid={dataTestid}
        />
      </LazyLoad>
    );
  };

  return <>{chooseImageComp()}</>;
};

export default ImageComp;
