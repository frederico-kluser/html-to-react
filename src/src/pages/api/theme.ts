import { connectToDatabase } from '../../database/connect';
import type { NextApiRequest, NextApiResponse } from 'next';

export const getTheme = async () => {
  const { db } = await connectToDatabase();
  return await db.collection('themes').findOne({ name: 'Light' });
};

const handler = async (req: NextApiRequest, res: NextApiResponse) => {
  const theme = await getTheme();
  res.status(200).json(theme);
};

export default handler;
