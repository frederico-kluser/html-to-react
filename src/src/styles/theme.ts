export const theme = {
  id: 'T1',
  name: 'Light',
  layout: {
    type: 'fixed',
    maxWidth: 1200,
    widthUnit: 'px'
  },
  colors: {
    primary: {
      main: '#0C1446',
      common: {
        black: '#000',
        white: '#fff',
        gray: '#616161',
        bgColor: '#af9995'
      }
    }
  },
  breakpoints: {
    size: {
      xs: 0,
      sm: 600,
      md: 960,
      lg: 1280,
      xl: 1920
    },
    unit: 'px'
  }
};
