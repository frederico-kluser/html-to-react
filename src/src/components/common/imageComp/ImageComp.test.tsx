import { render, screen, waitFor } from '@testing-library/react';
import { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../../styles/theme';
import ImageComp from '.';
import { forceVisible } from 'react-lazyload';

const renderComponent = (children: ReactNode) =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);

describe('<ImageComp />', () => {
  it('Should render the heading', async () => {
    const { container } = renderComponent(
      <ImageComp src="http://pudim.com.br/pudim.jpg" alt="pudim" />
    );
    forceVisible();
    await waitFor(() => expect(container).toBeInTheDocument());

    expect(screen.getByAltText(/pudim/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
