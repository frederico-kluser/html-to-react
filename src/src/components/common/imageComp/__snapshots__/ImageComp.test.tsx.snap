// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageComp /> Should render the heading 1`] = `
<div
  class="lazyload-wrapper "
>
  <img
    alt="pudim"
    src="http://pudim.com.br/pudim.jpg"
  />
</div>
`;
